lib/src/postgresql_impl/connection.dart

```diff
@@ -10,7 +10,7 @@ class ConnectionImpl implements Connection {
       TypeConverter typeConverter,
       String getDebugName())
     : _userName = settings.user,
-      _passwordHash = _md5s(settings.password + settings.user),
+      _passwordHash = _md5s(Uri.decodeFull(settings.password) + settings.user),
```

Comment:
Would this be better placed in the settings class? What about decoding the user name?